source/docs/software/wpilib-tools/wpical/index.rst

```diff
@@ -0,0 +1,177 @@
+
+* :guilabel:`Board Height` is the height (in squares) of the board
+
+For :guilabel:`OpenCV`, there is one more option to fill in:
+
+* :guilabel:`Marker Width` is the width of the ChArUco markers (in inches) on the board
+
+.. image:: images/OpenCV.png
+    :alt: OpenCV
+
+For :guilabel:`MRcal`, there are three more options to fill in:
+
+* :guilabel:`Image Width` is the width (in pixels) of the video's resolution
+* :guilabel:`Image Height` is the height (in pixels) of the video's resolution
+* :guilabel:`Focal Length` is the focal length (in pixels) of the camera.
+
+.. image:: images/MRcal.png
+    :alt: MRcal
+
+.. important:: For the calibration to work properly, it is important that you enter the exact width of the squares (and markers for ChArUco). It is recommended to use calipers to accurately measure these widths.
+
+When all the calibration fields have been entered, select :guilabel:`Calibrate` to run the calibration. A window will appear showing you the corner detection process:
+
+ChArUco:
+
+.. image:: images/ChArUcoDetection.png
+    :alt: ChArUcoDetection
+
+Checkerboard:
+
+.. image:: images/ChessboardDetection.png
+    :alt: CheckerboardDetection
+
+.. warning:: If any frames from the camera calibration look suspect, take a new video of the calibration board and try again.
+
+The camera calibration will automatically load the generated camera intrinsics JSON into WPIcal to continue to field calibration, but will also output the JSON file so it can be used for future calibrations.
+
+Upload Calibration JSON
+^^^^^^^^^^^^^^^^^^^^^^^
+
+As opposed to calibrating cameras in the tool, WPIcal also allows you to upload a camera intrinsics JSON. There are a three things each calibration JSON needs:
+
+* ``avg_reprojection_error``
+* ``camera_matrix``
+* ``distortion_coefficients``
+
+Example:
+
+.. code-block:: json
+    
+    {
+        "avg_reprojection_error": 0.3989609373420966,
+        "camera_matrix": [
+            1767.912766105077,          // fx
+            0.0,
+            989.4346078930205,          // cx
+            0.0,
+            1763.5161174120185,         // fy
+            540.7047848580446,          // cy
+            0.0,
+            0.0,
+            1.0
+        ],
+        "distortion_coefficients": [
+            0.3878712310541597,         // k1
+            -2.565991238351721,         // k2
+            -0.0011767048240595168,     // p1
+            0.0017341455009923185,      // p2
+            5.70562560518866,           // k3
+            0.03694246896004978,        // k4
+            0.11811204683668342,        // k5
+            -0.2274481545065017,        // k6
+            0.0,                        // s1
+            0.0,                        // s2
+            0.0,                        // s3
+            0.0,                        // s4
+            0.0,                        // tx
+            0.0                         // ty
+        ]
+    }
+
+
```

Comment:
The CalibDB section seems like it should be up higher, maybe a new section after in-app calibration